core/src/main/java/org/apache/carbondata/core/metadata/converter/ThriftWrapperSchemaConverterImpl.java

```diff
@@ -392,9 +392,9 @@ private DataType fromExternalToWrapperDataType(org.apache.carbondata.format.Data
       case DATE:
         return DataTypes.DATE;
       case ARRAY:
-        return DataTypes.ARRAY;
+        return DataTypes.createDefaultArrayType();
```

Comment:
Can't we create actual array from the  child schemas here?

Reply:
In schema.thrift, DataType is a Enum, so I think we can not get the child schema from existing table before this PR.